nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,840 @@
+                        range(n_problems), self.cv_))
+
+        coefs, intercepts = self._fetch_parallel_fit_outputs(
+                                    parallel_fit_outputs, y, n_problems)
+
+        # Build the final model (the aggregated one)
+        self.coef_ = np.vstack([np.mean(coefs[class_index], axis=0)
+                                for class_index in self.classes_])
+        self.std_coef_ = np.vstack([np.std(coefs[class_index], axis=0)
+                                   for class_index in self.classes_])
+        self.intercept_ = np.hstack([np.mean(intercepts[class_index], axis=0)
+                                    for class_index in self.classes_])
+
+        self.coef_img_, self.std_coef_img_ = self._output_image(
+            self.classes_, self.coef_, self.std_coef_)
+
+        if self.is_classification and (self.n_classes_ == 2):
+            self.coef_ = self.coef_[0, :][np.newaxis, :]
+            self.intercept_ = self.intercept_[0]
+
+    def decision_function(self, X):
+        """Predict class labels for samples in X.
+
+        Parameters
+        ----------
+        X: list of Niimg-like objects
+            See <http://nilearn.github.io/manipulating_images/input_output.html>
+            Data on prediction is to be made. If this is a list,
+            the affine is considered the same for all.
+
+        Returns
+        -------
+        y_pred: ndarray, shape (n_samples,)
+            Predicted class label per sample.
+        """
+        X = self.masker_.transform(X)
+
+        n_features = self.coef_.shape[1]
+        if X.shape[1] != n_features:
+            raise ValueError(
+                "X has {} features per sample; expecting {}".format(
+                              (X.shape[1], n_features)))
+
+        scores = safe_sparse_dot(X, self.coef_.T,
+                                 dense_output=True) + self.intercept_
+
+        return scores.ravel() if scores.shape[1] == 1 else scores
+
+    def predict(self, X):
+        """Predict a label for all X vectors indexed by the first axis.
+
+        Parameters
+        ----------
+        X: {array-like, sparse matrix}, shape = (n_samples, n_features)
+            Samples.
+
+        Returns
+        -------
+        array, shape=(n_samples,) if n_classes == 2 else (n_samples, n_classes)
+            Confidence scores per (sample, class) combination. In the binary
+            case, confidence score for self.classes_[1] where >0 means this
+            class would be predicted.
+        """
+
+        check_is_fitted(self, "coef_")
+        check_is_fitted(self, "masker_")
+
+        scores = self.decision_function(X)
+
+        if self.is_classification:
+            if len(scores.shape) == 1:
+                indices = (scores > 0).astype(np.int)
+            else:
+                indices = scores.argmax(axis=1)
+            return self.classes_[indices]
+
+        return scores
+
+    def _apply_mask(self, X, standardize=False):
```

Comment:
I do not see the masker parameters `fwhm`, `mask_strategy`, `low_pass`, `high_pass`, `t_r`, `target_shape`, `target_affine` being used anywhere. Am I missing something?

Reply:
These parameters are passed from init of the `_BaseDecoder` object to `check_embedded_nifti_masker(self, multi_subject=False)` just below to initialize `self.masker_` so that the masker has all these parameters set as desired.